File: POO python/practicas/tiempoprueba.py
from practicas.tiempo import Tiempo

t1 = Tiempo(10, 20, 30)  # Establecemos los valores de Tiempo

# Sumamos y restamos Horas
print(f"T1: {t1}")
h = int(input(f"Horas a sumar a {t1}"))
t1.suma_horas(h)
print(f"Ahora T1 es {t1}")
h = int(input(f"Horas a restar a {t1}"))
t1.resta_horas(h)
print(f"Ahora T1 es {t1}")

# Sumamos y restamos minutos
m = int(input(f"Minutos a sumar a {t1}"))
t1.suma_minutos(m)
print(f"Ahora T1 es {t1}")
m = int(input(f"Minutos a restar a {t1}"))
t1.resta_minutos(m)
print(f"Ahora T1 es {t1}")

# Sumamos y restamos segundos
s = int(input(f"Segundos a sumar a {t1}"))
t1.suma_segundos(s)
print(f"Ahora T1 es {t1}")
s = int(input(f"Segundos a restar a {t1}"))
t1.resta_segundos(s)
print(f"Ahora T1 es {t1}")

# Sumamos y restamos otro objeto de la clase Tiempo
print("Para sumar T2 a T1 introduce los valores de T2")
h = int(input("Horas de T2: "))
m = int(input("Minutos de T2: "))
s = int(input("Segundos de T2: "))
t2 = Tiempo(h, m, s)
t1.suma(t2)
print(f"Ahora T1 es {t1}")

print("Para restar T2 a T1 introduce los valores de T2")
h = int(input("Horas de T2: "))
m = int(input("Minutos de T2: "))
s = int(input("Segundos de T2: "))
t2 = Tiempo(h, m, s)
t1.resta(t2)
print(f"Ahora T1 es {t1}")

File: POO java/src/practica/TiempoPrueba.java
package practica;
import practica.Tiempo;
import java.util.Scanner;

public class TiempoPrueba {

	public static void main(String[] args) {
	int h;
	int m;
	int s;
  Tiempo t1 = new Tiempo(10,10,20);
  @SuppressWarnings("resource")
	Scanner scanner = new Scanner(System.in);
	
	System.out.println("T1: " + t1);
	System.out.println("Horas a sumar a " + t1);
  h = Integer.parseInt(scanner.nextLine());
	t1.sumaHoras(h);
	System.out.println("Ahora T1 es " + t1);
	System.out.println("Horas a restar a " + t1);
	h = Integer.parseInt(scanner.nextLine());
	t1.restaHoras(h);
	System.out.println("Ahora T1 es " + t1);
	
	System.out.println("T1: " + t1);
	System.out.println("Minutos a sumar a " + t1);
  m = Integer.parseInt(scanner.nextLine());
	t1.sumaMinutos(m);
	System.out.println("Ahora T1 es " + t1);
	System.out.println("Minutos a restar a " + t1);
	m = Integer.parseInt(scanner.nextLine());
	t1.restaMinutos(m);
	System.out.println("Ahora T1 es " + t1);

	System.out.println("T1: " + t1);
	System.out.println("Segundos a sumar a " + t1);
  s = Integer.parseInt(scanner.nextLine());
	t1.sumaSegundos(s);
	System.out.println("Ahora T1 es " + t1);
	System.out.println("Segundos a restar a " + t1);
	s = Integer.parseInt(scanner.nextLine());
	t1.restaSegundos(s);
	System.out.println("Ahora T1 es " + t1);

	System.out.println("Para sumar T2 a T1 introduce los valores de T2");
	System.out.println("Horas de T2: ");
  h = Integer.parseInt(scanner.nextLine());
	System.out.println("Minutos de T2: ");
  m = Integer.parseInt(scanner.nextLine());
	System.out.println("Segundos de T2: ");
  s = Integer.parseInt(scanner.nextLine());
	Tiempo t2 = new Tiempo(h, m, s);
	t1.suma(t2);
	System.out.println("Ahora T1 es " + t1);

	System.out.println("Para restar T2 a T1 introduce los valores de T2");
	System.out.println("Horas de T2: ");
  h = Integer.parseInt(scanner.nextLine());
	System.out.println("Minutos de T2: ");
  m = Integer.parseInt(scanner.nextLine());
	System.out.println("Segundos de T2: ");
  s = Integer.parseInt(scanner.nextLine());
	t2 = new Tiempo(h, m, s);
	t1.resta(t2);
	System.out.println("Ahora T1 es " + t1);
	}
}